src/ngx_http_lua_common.h

```diff
@@ -382,8 +382,10 @@ typedef struct ngx_http_lua_ctx_s {
 
     ngx_int_t                exit_code;
 
-    ngx_http_lua_co_ctx_t   *downstream_co_ctx; /* co ctx for the coroutine
-                                                   reading the request body */
+    ngx_http_lua_co_ctx_t   *downstream_read_co_ctx; /* co ctx for the coroutine
+                                                        reading the request body */
+    ngx_http_lua_co_ctx_t   *downstream_write_co_ctx; /* co ctx for the coroutine
```

Comment:
Hmm, seems like we can save some space by reusing the existing `read_co_ctx` and `write_co_ctx` fields here? We already have the `body_downstream` and `raw_downstream` flags to identify downstream sockets anyway :)